src/main/java/at/hannibal2/skyhanni/features/chat/PatternHiderManager.kt

```diff
@@ -0,0 +1,51 @@
+package at.hannibal2.skyhanni.features.chat
+
+import at.hannibal2.skyhanni.SkyHanniMod
+import at.hannibal2.skyhanni.utils.LorenzUtils
+import net.minecraft.client.Minecraft
+import net.minecraft.util.IChatComponent
+
+class PatternHiderManager(skyHanniMod: SkyHanniMod) {
+
+    private val referenceCounter = mutableListOf<Counter>()
+
+    init {
+        val watchdogHider = PatternHider(this, "watchdog", watchdogPattern) {
```

Comment:
We could make pattern Hider a data class and keep the logic methods and the forge events in the manager class